src/nvim/channel.h

```diff
@@ -130,5 +140,28 @@ static inline Stream *channel_outstream(Channel *chan)
   abort();
 }
 
+static inline Channel *acquire_asynccall_channel(void)
```

Comment:
as discussed in chat, I think we should avoid managing a pool. On a warm system spawning a new `nvim` costs ~10 milliseconds (as long as it doesn't load user config).